src/data/polynomial/basic.lean

```diff
@@ -334,6 +334,14 @@ begin
     rw [mul_assoc, X_mul, ←mul_assoc, ih, mul_assoc, ←pow_succ'], }
 end
 
+/--  This lemma is the loop-avoiding `simp` version of `X_mul`. -/
+@[simp] lemma X_mul_C {r : R} : X * C r = C r * X :=
+X_mul
+
+/--  This lemma is the loop-avoiding `simp` version of `X_pow_mul`. -/
+@[simp] lemma X_pow_mul_C {r : R} {n : ℕ} : X^n * C r = C r * X^n :=
+X_pow_mul
+
 lemma X_pow_mul_assoc {n : ℕ} : (p * X^n) * q = (p * q) * X^n :=
```

Comment:
It's probably sensible to add the `C` version of this one too:
```lean
lemma X_pow_mul_C-assoc {n : ℕ} : (p * X^n) * C r = (p * C r) * X^n :=
```

Reply:
I added this new lemma as well!  Thanks!

Reply:
Although now I wonder whether the `r : R` in this version should really be a `p : R[X]`.